hummingbot/connector/exchange/bitmax/bitmax_order_book_message.py

```diff
@@ -0,0 +1,79 @@
+#!/usr/bin/env python
+
+from typing import (
+    Dict,
+    List,
+    Optional,
+)
+
+from hummingbot.core.data_type.order_book_row import OrderBookRow
+from hummingbot.core.data_type.order_book_message import (
+    OrderBookMessage,
+    OrderBookMessageType,
+)
+
+
+class BitmaxOrderBookMessage(OrderBookMessage):
+    def __new__(
+        cls,
+        message_type: OrderBookMessageType,
+        content: Dict[str, any],
+        timestamp: Optional[float] = None,
+        *args,
+        **kwargs,
+    ):
+        if timestamp is None:
+            if message_type is OrderBookMessageType.SNAPSHOT:
+                raise ValueError("timestamp must not be None when initializing snapshot messages.")
+            timestamp = content["timestamp"]
+
+        return super(BitmaxOrderBookMessage, cls).__new__(
+            cls, message_type, content, timestamp=timestamp, *args, **kwargs
+        )
+
+    @property
+    def update_id(self) -> int:
+        if self.type in [OrderBookMessageType.DIFF, OrderBookMessageType.SNAPSHOT]:
+            # return int(self.content["seqnum"])
+            return int(self.timestamp)
+        else:
+            return -1
+
+    @property
+    def trade_id(self) -> int:
+        if self.type is OrderBookMessageType.TRADE:
+            # return int(self.content["seqnum"])
```

Comment:
Should we be using `seqnum` for `trade_id` considering that

> `seqnum` is always increasing for each symbol 
>  `ts` is the UTC timestamp in milliseconds of the **first** trade

I am just concerned that there might be multiple trades with the same ids

Reply:
I will need some time to test this out, for now, I will stick with timestamp